.github/workflows/build_mpich_source.yml

```diff
@@ -53,8 +56,8 @@ jobs:
 
       # Enable threads=multiple for testing with Subfiling and
       # VOL connectors that require MPI_THREAD_MULTIPLE
-      - name: Install MPICH (GCC) (Production)
-        if: ${{ steps.cache-mpich-ubuntu-gcc.cache-hit != 'true' && (inputs.build_mode != 'debug') }}
```

Comment:
Fixed the caching here in both the MPICH and OpenMPI workflows; forgot to add the `.outputs`